core/core-impl/src/test/java/ai/timefold/solver/core/impl/testdata/util/PlannerTestUtils.java

```diff
@@ -106,14 +107,14 @@ public static TestdataSolution generateTestdataSolution(String code, int entityA
     // ScoreDirector methods
     // ************************************************************************
 
-    public static <Solution_> InnerScoreDirector<Solution_, SimpleScore> mockScoreDirector(
+    public static <Solution_> AbstractScoreDirector<Solution_, SimpleScore, ?> mockScoreDirector(
             SolutionDescriptor<Solution_> solutionDescriptor) {
         EasyScoreDirectorFactory<Solution_, SimpleScore> scoreDirectorFactory =
                 new EasyScoreDirectorFactory<>(solutionDescriptor, (solution_) -> SimpleScore.of(0));
         scoreDirectorFactory.setInitializingScoreTrend(
                 InitializingScoreTrend.buildUniformTrend(InitializingScoreTrendLevel.ONLY_DOWN, 1));
-        return mock(InnerScoreDirector.class,
-                AdditionalAnswers.delegatesTo(scoreDirectorFactory.buildScoreDirector(false, false)));
+        return mock(AbstractScoreDirector.class,
+                AdditionalAnswers.delegatesTo(scoreDirectorFactory.buildScoreDirector(true, false)));
```

Comment:
Any particular reason to enable lookup in the mock?

Reply:
Yes. See `AbstractScoreDirector.initializeElements()`.

When a phase ends and a new one starts, the score director is reset, resetting the initialization state of elements. This means I need to remember it before the reset happens, and restore it after the reset happens. Unfortunately, this means everything will be cloned in the meantime; so I need a lookup.

Now that I describe it, I realize though that this is not acceptable. This effectively means the solver can not live without lookup enabled, which is arguably not what we need. I'll have to rethink this.